buildpacks/go/src/layers/dist.rs

```diff
@@ -45,7 +45,10 @@ impl Layer for DistLayer {
         _ctx: &BuildContext<Self::Buildpack>,
         layer_path: &Path,
     ) -> Result<LayerResult<Self::Metadata>, GoBuildpackError> {
-        log_info(format!("Installing Go {}", self.artifact.semantic_version));
+        log_info(format!(
+            "Installing Go {} from {}",
+            self.artifact.semantic_version, self.artifact.url
```

Comment:
Should this be consistent with the message when the layer is restored?  E.g.; 

##### Cache Created
```
Installing Go 1.22.1 from https://go.dev/dl/go1.22.1.linux-amd64.tar.gz
``` 

##### Cache Restored
```
Reusing Go 1.22.1 (linux-x86_64)
```

There is a bit of overlap in the output since the url does contain some OS/Arch information but it's slightly different (i.e.; `x86_64` vs. `amd64`) so it might be worthwhile if both cases are more aligned.

##### Cache Created
```
Installing Go 1.22.1 (linux-x86_64) from https://go.dev/dl/go1.22.1.linux-amd64.tar.gz
``` 

```suggestion
            self.artifact, self.artifact.url
```

Reply:
Agreed, this is more consistent. Was actually planning on making that change after this PR, especially since using the `semantic_version` in user output doesn't seem ideal in the first place (it's a bit of an implementation related to how we resolve the artifact to install, and doesn't always map directly to an upstream Go version (e.g. there is no `go1.20.0`)) -- but I've included the change anyway since you asked https://github.com/heroku/buildpacks-go/pull/237/commits/f47ae360a8928aeb81fd684bdccf508a72006185!

Also yes the information is duplicated a bit in the URL. It does seem preferable to tell users where exactly we're getting the Go version from, and it has the added benefit of "mapping" our `Artifact` arch values to those used in Go (and reduce the risk that using slightly different values will cause confusion for users) - it didn't sound like you had any objections to that though :)